src/neo/Network/P2P/Peer.cs

```diff
@@ -208,25 +211,27 @@ private void OnStart(ChannelsConfig config)
         private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
         {
             ImmutableInterlocked.Update(ref ConnectingPeers, p => p.Remove(remote));
+
+            // Pre-check includes MaxConnections, MaxConnectionsPerAddress. If the check fails, it'll seed the error message.
             if (MaxConnections != -1 && ConnectedPeers.Count >= MaxConnections && !TrustedIpAddresses.Contains(remote.Address))
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                Sender.Tell(new Tcp.Register(ActorRefs.Nobody));
```

Comment:
Why do you send this message?

Reply:
> To create a connection an actor sends a Tcp.Connect message to the TCP Manager. Once the connection is established the connection actor sends a Tcp.Connected message to the commander, which registers the connection handler by replying with a Tcp.Register message. https://getakka.net/articles/networking/io.html#getting-started

We need to register the handler, otherwise we can't send the error message back.

Reply:
Get it. But then it should be sent in `LocalNode`. Because the `Peer` doesn't send the message when disconnecting.

Reply:
No, we also need to send the register message in normal case#L233

```cs
private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
    ........
	ConnectedAddresses[remote.Address] = count + 1;
	IActorRef connection = Context.ActorOf(ProtocolProps(Sender, remote, local), $"connection_{Guid.NewGuid()}");
	Context.Watch(connection);
	Sender.Tell(new Tcp.Register(connection));  <------ here
	ConnectedPeers.TryAdd(connection, remote);
}
```

Reply:
But we don't need to send `Register` when it is disconnecting.

Reply:
As we want to send the disconnection reason attched with peers list.

Reply:
The disconnection reason is sent in `LocalNode`.

Reply:
Yes, we send reason message in LocalNode, and create reason in Peer, as we have abstract the `TcpDisconnect`.  I think it's okay to put `Sender.Tell(new Tcp.Register(connection));` here, which  is part of the akka handshake protocol.

Reply:
It's illogical that register an actor and then close it immediately.

Reply:
We also want a better way, like HTTP various status codes, but akka doesn't support it, and we also want to carry the address list information.

Reply:
@erikzhang, perhaps it is reaching the time to merge this PR for us to start testing it together with other daily routines.